scopes/typescript/typescript/typescript.extractor.ts

```diff
@@ -52,7 +52,19 @@ export class TypeScriptExtractor implements SchemaExtractor {
   /**
    * extract a component schema.
    */
-  async extract(component: Component, formatter?: Formatter): Promise<APISchema> {
+  async extract(
+    component: Component,
+    formatter?: Formatter,
+    overrideRootTsserverPath?: string,
+    overrideContextPath?: string
+  ): Promise<APISchema> {
+    // override the rootTsserverPath and rootContextPath if passed
+    if (overrideRootTsserverPath) {
+      this.rootTsserverPath = overrideRootTsserverPath;
+    }
+    if (overrideContextPath) {
+      this.rootContextPath = overrideContextPath;
+    }
```

Comment:
why do you need to override the members and not just using the paths you get?

Reply:
because then anywhere the path is accessed via instance of the extractor, it will result in a different path and then I need to pass this override everywhere